openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermModal/GlossaryTermModal.component.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright 2023 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { Button, Modal } from 'antd';
+import { useForm } from 'antd/lib/form/Form';
+import AddGlossaryTermForm from 'components/AddGlossaryTermForm/AddGlossaryTermForm.component';
+import { GlossaryTermForm } from 'components/AddGlossaryTermForm/AddGlossaryTermForm.interface';
+import { GlossaryTerm } from 'generated/entity/data/glossaryTerm';
+import { EntityReference } from 'generated/type/entityReference';
+import React, { FC, useEffect, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface Props {
+  glossaryName: string;
+  glossaryTerm: GlossaryTerm | undefined;
+  glossaryReviewers?: EntityReference[];
+  onSave: (value: GlossaryTermForm) => void;
+  onCancel: () => void;
+  visible: boolean;
+  editMode: boolean;
+}
+
+const GlossaryTermModal: FC<Props> = ({
+  editMode,
+  glossaryName,
+  visible,
+  glossaryTerm,
+  glossaryReviewers = [],
+  onSave,
+  onCancel,
+}) => {
+  const { t } = useTranslation();
+  const [form] = useForm();
+
+  const dialogTitle = useMemo(() => {
+    return editMode
+      ? t('label.edit-entity', { entity: t('label.glossary-term') })
+      : t('label.add-entity', { entity: t('label.glossary-term') });
+  }, [editMode]);
+
+  useEffect(() => {
+    !visible && form.resetFields();
+  }, [visible]);
+
+  return (
+    <Modal
+      centered
+      destroyOnClose
+      cancelText={t('label.cancel')}
+      className="edit-glossary-modal"
+      closable={false}
+      data-testid="edit-glossary-modal"
+      footer={[
+        <Button key="cancel-btn" type="link" onClick={onCancel}>
+          {t('label.cancel')}
+        </Button>,
+        <Button
+          data-testid="save-glossary-term"
+          key="save-btn"
+          type="primary"
+          onClick={() => form.submit()}>
+          {t('label.save')}
+        </Button>,
+      ]}
+      maskClosable={false}
+      okText={t('label.save')}
+      open={visible}
+      title={dialogTitle}
+      width={800}>
+      <AddGlossaryTermForm
+        allowAccess
+        isFormInModal
```

Comment:
Can't see benefit of it unless we are using this form in page too?

Reply:
Kept it for now as we have not removed page component.